submissions/Vitamin/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,100 @@
+const Net = require('net');
+const dns = require('dns').promises;
+const { help, isDefined } = require('./helpers');
+
+const [, , ...processArgs] = process.argv;
+const [minPortValue, maxPortValue] = [0, 65535];
+
+const ipToURL = async host => {
+  try {
+    const result = await dns.lookup(host);
+    return await result.address;
+  } catch (err) {
+    throw Error('Invalid host');
+  }
+};
+
+const isPortInRange = port => port >= minPortValue && port <= maxPortValue;
+
+const getPortsRange = ports => {
+  const [firstPort, last] = ports.split('-');
+  const lastPort = isPortInRange(last) ? last : firstPort;
+  if (isPortInRange(firstPort)) {
+    return [firstPort, lastPort];
+  }
+  // eslint-disable-next-line no-console
+  console.log('Invalid ports range');
+  return undefined;
+};
+
+const tryToConnect = (host, port) => {
+  const client = new Net.Socket();
+  client.setTimeout(300);
+  client.connect({ port, host });
+  return new Promise(resolve => {
+    client.on('connect', () => {
+      process.stdout.write('.');
+      client.destroy();
+      return resolve(port);
+    });
+    client.on('timeout', () => {
+      client.destroy();
+      return resolve(false);
```

Comment:
I would prefer to see `null` as sign of missing value. In this case you can filter only by one condition (non equal to null).

Reply:
Excellent.
How I haven't figured out that by myself :)